client/app/application.coffee

```diff
@@ -2,7 +2,37 @@ module.exports =
 
     listenTo: Backbone.Model.prototype.listenTo
 
-    initialize: ->
+
+    # Initialize a custom error handler for the global app
+    initializeErrorHandler: (window) ->
+        existingDefaultHandler = window.onerror
+
+        applicationErrorHandler = (msg, url, line, col, error) ->
+            # Handler for asynchronous errors
+            @onEventSharingError = (error) ->
+                # TODO find a better way to format a string like this
+                alert [ t('Event sharing failed for event'),
```

Comment:
Localization keys should be composed of lowercase words and describe the context of the text.